### ISSUE TITLE
Performence issue with large elm files

### ISSUE THREAD

**Issue by miltonio94** [jcollard/elm-mode#182]
Hello,
I've been having some big performance issues with elm mode in large files. I have noticed it be particularly bad in one file in specific that is 11737 lines long. The weird thing is that it doesn't happen at all times, but the auto-indenting completely stalls emacs for seconds. I have not been able to identify what triggers it. At first, I thought it was because of me being on a zoom call or a slack call, but I get the problem even when I'm not under those circumstances.
My work machine (where I have been having this issue) is no slouch either. It has an i7 and 42Gb of RAM.
I use this package with doom emacs at the moment.

I'm new to writing issues so if anything isn't up to scratch, let me know.

**Comment by purcell** [comment]
Is that auto-indenting with elm-format (e.g. when you save), or you mean using the Emacs commands (e.g. mark whole buffer, reindent)?

**Comment by miltonio94** [comment]
yeah, on save with elm-format, it also happens when i run just the command. It keeps giving me a timeout error. I thought this was due to the size of the file, but when i run the elm-format command from the terminal it runs it fine on these large files in less than a second without any hangup. (sorry it took some time to reply, i missed the notification)

**Comment by purcell** [comment]
Thanks. Would you mind sharing which emacs version you're using?

**Comment by miltonio94** [comment]
I found the issue originally on emacs 27, I'm currently using 29.0.5 which has had a slight performance boost for me
